pyispyb/app/extensions/database/definitions.py

```diff
@@ -70,3 +41,103 @@ def get_current_person(login: str) -> Optional[models.Person]:
     person._metadata["permissions"] = permissions
 
     return person
+
+
+def with_authorization(
+    query: "sqlalchemy.orm.Query[Any]",
+    includeArchived: bool = False,
+    proposalColumn: "sqlalchemy.Column[Any]" = None,
+    joinBLSession: bool = True,
```

Comment:
Maybe have `joinBLSession=False` as default to be consistent with `proposalColumn=None`.
This way, default behavior is only check for authorization, and if you need some joins to be made for you, you have specify it.

Reply:
Also I think most of the calls are actually made with `joinBLSession=False`.

Reply:
Yer thats because most of these are datacollection related, there will be a big chunk without too (lab contacts, samples, shipping, containers, etc)